lib/codegen/recorder.dart

```diff
@@ -257,7 +257,8 @@ class Recorder {
     // [node] is the initialization expression, we walk up to get to the actual
     // member declaration where the metadata is attached to.
     while (node is! ClassMember) node = node.parent;
-    return node.metadata.map(_convertAnnotation).toList();
+    return (node as ClassMember).metadata.map(_convertAnnotation).toList() as
```

Comment:
I believe we are already there :-) (see [map definition][1]). So we can write this as:
```dart
return (node as ClassMember).metadata.map/*<ConstExpression>*/(_convertAnnotation).toList()
```

Note that here the cast alone has a similar problem like the one I mentioned earlier: `toList() as List<ConstExpression>` will fail with a runtime cast. This case can work without using generics too. You could use: `new List.from(...toList())`. In this case DDC can infer from the return type that List.from's type argument should be ConstExpression, so it's OK to write `new List.from(...)` instead of `new List.from<ConstExpression>(...)`. The main difference is that `new List.from` will have an extra cast/check while initializing the list, but in the case of the generic-map call, those checks could be optimized away (nothing today does that, but it's a possible optimization in the future)


[1]:https://github.com/dart-lang/sdk/blob/master/sdk/lib/core/iterable.dart#L159